File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/port/ConversationSessionRepository.java
package com.study.webflux.rag.domain.dialogue.port;

import com.study.webflux.rag.domain.dialogue.model.ConversationSession;
import com.study.webflux.rag.domain.dialogue.model.ConversationSessionId;
import com.study.webflux.rag.domain.dialogue.model.PersonaId;
import com.study.webflux.rag.domain.dialogue.model.UserId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ConversationSessionRepository {

	Mono<ConversationSession> save(ConversationSession session);

	Mono<ConversationSession> findById(ConversationSessionId sessionId);

	Flux<ConversationSession> findByUserId(UserId userId);

	Flux<ConversationSession> findByPersonaId(PersonaId personaId);

	Flux<ConversationSession> findByPersonaIdAndUserId(PersonaId personaId, UserId userId);

	Mono<ConversationSession> softDelete(ConversationSessionId sessionId);
}

File: domain/src/main/kotlin/com/miyou/app/domain/dialogue/port/ConversationSessionRepository.kt
package com.miyou.app.domain.dialogue.port

import com.miyou.app.domain.dialogue.model.ConversationSession
import com.miyou.app.domain.dialogue.model.ConversationSessionId
import com.miyou.app.domain.dialogue.model.PersonaId
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface ConversationSessionRepository {
    fun save(session: ConversationSession): Mono<ConversationSession>

    fun findById(sessionId: ConversationSessionId): Mono<ConversationSession>

    fun findByUserId(userId: String): Flux<ConversationSession>

    fun findByPersonaId(personaId: PersonaId): Flux<ConversationSession>

    fun findByPersonaIdAndUserId(
        personaId: PersonaId,
        userId: String,
    ): Flux<ConversationSession>

    fun softDelete(sessionId: ConversationSessionId): Mono<ConversationSession>
}
